protocols/virtualfund/virtualfund.go

```diff
@@ -49,6 +50,64 @@ func (c *Connection) Equal(d *Connection) bool {
 
 }
 
+// jsonConnection is a serialization-friendly struct representation
+// of a Connection
+type jsonConnection struct {
+	Channel            types.Destination
+	ExpectedGuarantees []assetGuarantee
+}
+
+func (c Connection) MarshalJSON() ([]byte, error) {
+	guarantees := []assetGuarantee{}
+	for asset, guarantee := range c.ExpectedGuarantees {
+		guarantees = append(guarantees, assetGuarantee{
+			asset,
+			guarantee,
+		})
+	}
+	jsonC := jsonConnection{c.Channel.Id, guarantees}
+	bytes, err := json.Marshal(jsonC)
+
+	if err != nil {
+		return []byte{}, err
+	}
+
+	return bytes, err
+}
+
+func (c *Connection) UnmarshalJSON(data []byte) error {
+	c.Channel = &channel.TwoPartyLedger{}
+	c.ExpectedGuarantees = make(map[types.Address]outcome.Allocation)
+
+	if string(data) == "null" {
+		// populate a well-formed but blank-addressed Connection
+		c.Channel.Id = types.Destination{}
+		return nil
+	}
+
+	var jsonC jsonConnection
+	err := json.Unmarshal(data, &jsonC)
+
+	if err != nil {
+		return err
+	}
+
+	c.Channel.Id = jsonC.Channel
```

Comment:
This could be slightly confusing if you're not aware of how `UnmarshalJSON` works. You would get back connection with a `nil` channel that has the correct `Id`. This is probably fine if it's only the `Store` dealing with marshalling/unmarshalling

Reply:
Yes, intent here is that this is a utility method specifically for the `store`. (A wart, but the persisting of private fields sort of requires it).